DevAcademy/Services/DataService.swift

```diff
@@ -0,0 +1,42 @@
+//
+//  DataService.swift
+//  DevAcademy
+//
+//  Created by Tomáš Melichar on 18.07.2023.
+//
+
+import Foundation
+
+private enum ServiceError: Error {
+    case noData
+    case fatalError
+}
+
+final class DataService {
+    static let shared = DataService()
+    
+    var data: Result<Features, Error>?
+    
+    private init() {}
+    
+    func fetchData(closure: @escaping (Result<Features, Error>) -> ()) throws {
+        Timer.scheduledTimer(withTimeInterval: 2.2, repeats: false) { [weak self] _ in
+            if let data = self?.data {
+                closure(data)
+                return
+            } else {
+                closure(.failure(ServiceError.fatalError))
+            }
+        }
+    }
```

Comment:
✳️ Všimněte si, že vaše funkce `fetchData(closure:) throws` nikde "nehází" žádnou chybu. Vaši výjimku vkládáte do typu `Result`.
```
enum Result<Success, Failure: Error> {
  case success(Success)
  case failure(Failure)
}
```
⚠️ Před spuštěním `Timeru` zkontrolujte, zda-li `self.data` obsahuje nějaká data. Pokud obsahuje data, zavolejte `closure` rovnou a skončete vykonávání funkce. Pokud obsahuje `nil`, spustěte timer.
⚠️ Ve Closure kterou timer volá nastavte do proměnné `self?.data` vaše namockovaná data.

Reply:
✳️ - smazáno `throws`, když nehází chybu. :) 

Upraveno - snad je to už, dle zadání :)